### PR TITLE
Soup

### PR DESCRIPTION
**Pull request by LucasTrg**
This is wildy undocumented code.
Don't push it to prod.

I'm importing BS4 because i'm too lazy to do regexes in our case. 
Importing a script to curl instead of using python's requests as they get 403s and i don't want to figure out why

### CODE REVIEW

File: roboclic.py [hunk 1]
```diff
@@ -144,6 +147,60 @@ def increment_stats(updated_user, stats_file):
...
 
+def soup(update, context):
+    """
+    Uses bs4 and a curl script to scrape FLEP's daily menus and output all meals fitting the criterion.
+    As of know, only supports price threshold (/soup 10 returns all meal under 10 chf)
+    """    
+    text = ""
+
+    #Fetches the daily menu
+    os.system("sh script.sh {}".format(
```
Comment (Robb-Fr) [inline roboclic.py]: Mdrrr no way de laisser ce genre de possibility a un bot qui prend des user input oklm. Tu peux pas réécrire le script en Python?

File: roboclic.py [hunk 2]
```diff
@@ -144,6 +147,60 @@ def increment_stats(updated_user, stats_file):
...
+        datetime.datetime.today().strftime("%Y-%m-%d"))) 
+    f = open("menu.html", "r")
+
+    soup = BeautifulSoup(f, "html.parser")
+    menu = soup.find_all("tr", {"class": "menuPage"})
+
+    #Removes non-Lausanne results
+    excluded = set(["La Ruch", "Microci", "Hodler"])
```
Comment (Robb-Fr) [inline roboclic.py:175]: J'aurai whiteliste plutôt que black liste nn?

File: roboclic.py (hunk 1 above)
Comment (Robb-Fr) [reply roboclic.py]: Le script est safe mais l'accès à l'exécution du script me fait peur, bot pas dockerisé en plus iirc?

**Comment by Robb-Fr** [comment]
kill me plz

**Comment by LucasTrg** [comment]
je suis en train d'essayer de faire une version regex si vous préférez

**Comment by LucasTrg** [comment]
Mais ça pourrait être future proof d'avoir BS4 si on a envie de faire plus de projets de scraping

**Comment by LucasTrg** [comment]
You can't parse [X]HTML with regex. Because HTML can't be parsed by regex. Regex is not a tool that can be used to correctly parse HTML. As I have answered in HTML-and-regex questions here so many times before, the use of regex will not allow you to consume HTML. Regular expressions are a tool that is insufficiently sophisticated to understand the constructs employed by HTML. HTML is not a regular language and hence cannot be parsed by regular expressions. Regex queries are not equipped to break down HTML into its meaningful parts. so many times but it is not getting to me. Even enhanced irregular regular expressions as used by Perl are not up to the task of parsing HTML. You will never make me crack. HTML is a language of sufficient complexity that it cannot be parsed by regular expressions. Even Jon Skeet cannot parse HTML using regular expressions. Every time you attempt to parse HTML with regular expressions, the unholy child weeps the blood of virgins, and Russian hackers pwn your webapp. Parsing HTML with regex summons tainted souls into the realm of the living. HTML and regex go together like love, marriage, and ritual infanticide. The <center> cannot hold it is too late. The force of regex and HTML together in the same conceptual space will destroy your mind like so much watery putty. If you parse HTML with regex you are giving in to Them and their blasphemous ways which doom us all to inhuman toil for the One whose Name cannot be expressed in the Basic Multilingual Plane, he comes. HTML-plus-regexp will liquify the n​erves of the sentient whilst you observe, your psyche withering in the onslaught of horror. Rege̿̔̉x-based HTML parsers are the cancer that is killing StackOverflow it is too late it is too late we cannot be saved the transgression of a chi͡ld ensures regex will consume all living tissue (except for HTML which it cannot, as previously prophesied) dear lord help us how can anyone survive this scourge using regex to parse HTML has doomed humanity to an eternity of dread torture and security holes using regex as a tool to process HTML establishes a breach between this world and the dread realm of c͒ͪo͛ͫrrupt entities (like SGML entities, but more corrupt) a mere glimpse of the world of reg​ex parsers for HTML will ins​tantly transport a programmer's consciousness into a world of ceaseless screaming, he comes, the pestilent slithy regex-infection wil​l devour your HT​ML parser, application and existence for all time like Visual Basic only worse he comes he comes do not fi​ght he com̡e̶s, ̕h̵i​s un̨ho͞ly radiańcé destro҉ying all enli̍̈́̂̈́ghtenment, HTML tags lea͠ki̧n͘g fr̶ǫm ̡yo​͟ur eye͢s̸ ̛l̕ik͏e liq​uid pain, the song of re̸gular exp​ression parsing will exti​nguish the voices of mor​tal man from the sp​here I can see it can you see ̲͚̖͔̙î̩́t̲͎̩̱͔́̋̀ it is beautiful t​he final snuffing of the lie​s of Man ALL IS LOŚ͖̩͇̗̪̏̈́T ALL I​S LOST the pon̷y he comes he c̶̮omes he comes the ich​or permeates all MY FACE MY FACE ᵒh god no NO NOO̼O​O NΘ stop the an​*̶͑̾̾​̅ͫ͏̙̤g͇̫͛͆̾ͫ̑͆l͖͉̗̩̳̟̍ͫͥͨe̠̅s ͎a̧͈͖r̽̾̈́͒͑e n​ot rè̑ͧ̌aͨl̘̝̙̃ͤ͂̾̆ ZA̡͊͠͝LGΌ ISͮ̂҉̯͈͕̹̘̱ TO͇̹̺ͅƝ̴ȳ̳ TH̘Ë͖́̉ ͠P̯͍̭O̚​N̐Y̡ H̸̡̪̯ͨ͊̽̅̾̎Ȩ̬̩̾͛ͪ̈́̀́͘ ̶̧̨̱̹̭̯ͧ̾ͬC̷̙̲̝͖ͭ̏ͥͮ͟Oͮ͏̮̪̝͍M̲̖͊̒ͪͩͬ̚̚͜Ȇ̴̟̟͙̞ͩ͌͝S̨̥̫͎̭ͯ̿̔̀ͅ

**Comment by huguesdevimeux** [comment]
énorme

File: roboclic.py [hunk 3]
```diff
@@ -159,10 +162,10 @@ def soup(update, context):
     As of know, only supports price threshold (/soup 10 returns all meal under 10 chf)
     """
     text = ""
-
-    # Fetches the daily menu
-    os.system("sh script.sh {}".format(
-        datetime.datetime.today().strftime("%Y-%m-%d")))
+    if (datetime.now()-REQUEST_TIMER.get("soup", "1700-01-01T00:00:00Z GMT")).total_seconds()>=3600:
+        # Fetches the daily menu
+        os.system("sh script.sh {}".format(
```
Comment (Robb-Fr) [inline roboclic.py:167]: juste plus qu'à remplacer ca par ton curl directement et c'est stylé

File: roboclic.py [hunk 4]
```diff
@@ -159,10 +162,10 @@ def soup(update, context):
...
+            datetime.datetime.today().strftime("%Y-%m-%d")))
     f = open("menu.html", "r")
```
Comment (Robb-Fr) [inline roboclic.py:169]: on mettrait pas un `with`? enfin le contexte et tout se ferme bien comme ca?

File: script.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+curl -k "https://www.epfl.ch/campus/restaurants-shops-hotels/fr/offre-du-jour-de-tous-les-points-de-restauration/" -o menu.html
```
Comment (Robb-Fr) [inline script.sh:1]: Est ce qu'on garderait pas plutot sur le disque un json du truc processed? comme ca on reprocess pas a chaque query c'est smooth

**Review by Robb-Fr — COMMENTED** [review]
quelques commentaires

File: roboclic.py (hunk 3 above)
Comment (Robb-Fr) [reply roboclic.py:167]: Genre plutot que de call un script, call directement curl

File: roboclic.py (hunk 1 above)
Comment (LucasTrg) [reply roboclic.py]: Je connais pas assez bien les vulnérabilités mais on peut interdire l'écriture sur le script dans les ACLs

File: roboclic.py (hunk 2 above)
Comment (LucasTrg) [reply roboclic.py:175]: Nan, c'est plutot pratique pour les foodtrucks qui changent, etc...

C'est plus future proof je trouve. Au pire il y a quelques results en trop mais au moins il manque rien

File: roboclic.py (hunk 3 above)
Comment (LucasTrg) [reply roboclic.py:167]: on peut. J'aurais bien aimé de pouvoir trouver un moyen safe de call des scripts au futur

File: roboclic.py (hunk 4 above)
Comment (LucasTrg) [reply roboclic.py:169]: oui

File: script.sh (hunk 5 above)
Comment (LucasTrg) [reply script.sh:1]: BS4 est étonnament efficient ass fuck, mais on peut oui. 
L'idée pourrait être de modulariser le fetch de page into des jsons pour donner une interface chouette à de futurs scripts

**Pull request merged**